signer/src/transaction_signer.rs

```diff
@@ -425,6 +535,125 @@ where
         Ok(())
     }
 
+    fn create_new_state_machine(&mut self) -> Result<SignerStateMachine, error::Error> {
+        let signers: hashbrown::HashMap<u32, _> = self
```

Comment:
Yeah this is just because the `hashbrown::HashMap` leaks through the `wsts::signer_state::PublicKeys` struct. I would love to avoid taking in this dependency if we could.